web/html/src/manager/virtualization/guests/list/guests-list.js

```diff
@@ -0,0 +1,394 @@
+const React = require("react");
+const PropTypes = React.PropTypes;
+const MessageContainer = require("../../../../components/messages").Messages;
+const MessagesUtils = require("../../../../components/messages").Utils;
+const {Table, Column, SearchField, Highlight} = require("../../../../components/table");
+const Functions = require("../../../../utils/functions");
+const Utils = Functions.Utils;
+const {AsyncButton} = require("../../../../components/buttons");
+const {DeleteDialog} = require("../../../../components/dialog/DeleteDialog");
+const {DangerDialog} = require("../../../../components/dialog/DangerDialog");
+const {ModalButton} = require("../../../../components/dialog/ModalButton");
+const Systems = require("../../../../components/systems");
+import VirtualizationGuestActionApi from '../virtualization-guest-action-api';
+import VirtualizationGuestsListRefreshApi from '../virtualization-guests-list-refresh-api';
+
+class GuestsList extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      actionsResults: {},
+      errors: [],
+      selectedItems: []
+    }
+
+    this.onBeforeUnload = (event) => {
+        this.setState({
+            pageUnloading: true
+        });
+    };
+
+    this.handleSelectItems = (items) => {
+      this.setState({
+        selectedItems: items
+      });
+    };
+
+    this.selectGuest = (row) => {
+      this.setState({
+        selected: row
+      });
+    };
+
+    this.actionCallback = (results) => {
+      var actions = this.state.actionsResults;
+      Object.entries(results).forEach(([uuid, actionid]) => {
+        actions[uuid] = {'id': actionid, 'result': 'Queued'};
+      });
+      this.setState({actionsResults: actions});
```

Comment:
yes, you are right! `var actions = this.state.actionsResults` isn't doing anything here.

You can either use `Object.assign({}, this.state.actionsResults)` or the JS destructuring  `{...this.state.actionsResults}` 

I prefer the last one, although it might seem weird at first it's really powerful and clean. Be aware that both options do a shallow copy and not a deep one.

Anyway I would do something like this:

```javascript
      var newActions = {};
      Object.entries(results).forEach(([uuid, actionid]) => {
        newActions[uuid] = {'id': actionid, 'result': 'Queued'};
      });
      this.setState({actionsResults: {...this.state.actionsResults, ...newActions});
 //     this.setState({actionsResults: Object.assign({}, this.state.actionsResults, newActions)); // other option
```